lib/net.c

```diff
@@ -1578,7 +1684,9 @@ finalize_stage2(__attribute__((unused)) void *arg)
 		if (ret < 0)
 			return ret;
 	}
-	return 0;
+	if (change_logs_permissions() < 0)
+		return -1;
+	return drop_privileges(config.user);
```

Comment:
If no user is set in the configuration file, that is `pw_uid == 0`, we should do nothing here.